### PR TITLE
fix mac r install and windows python build from source docs

### PR DESCRIPTION
**Pull request by aaronmarkham**
## Description ##
Fixes https://github.com/apache/incubator-mxnet/issues/12892

* macOS R install section was not rendering properly. The rendering of this page is really brittle. I'm not sure why moving the dependencies down a few lines mattered, but that was the fix.
* I also noticed that the build from source for Python / Windows was missing/replaced by Docker, so I fixed that too.

## Preview
http://34.201.8.176/versions/install_macos_r_fix/install/index.html?platform=MacOS&language=R&processor=CPU

http://34.201.8.176/versions/install_macos_r_fix/install/index.html?platform=Windows&language=Python&processor=CPU - then select build from source

### CODE REVIEW

**Comment by aaronmarkham** [comment]
@JiaxiangBU - please let me know if this resolved the issue. You can try the preview links in the main PR description.

**Comment by roywei** [comment]
@aaronmarkham  Thanks for the contribution!
@mxnet-label-bot [pr-awaiting-review]

**Comment by aaronmarkham** [comment]
@ankkhedia @anirudhacharya - can you review?

File: docs/install/index.md [hunk 1]
```diff
@@ -620,6 +609,19 @@ options(repos = cran)
 install.packages("mxnet")
 ```
 
+To run MXNet you also should have OpenCV and OpenBLAS installed. You may install them with `brew` as follows:
```
Comment (anirudhacharya) [inline docs/install/index.md]: these steps need to be done before calling `install.packages`

**Comment by JiaxiangBU** [comment]
> @JiaxiangBU - please let me know if this resolved the issue. You can try the preview links in the main PR description.

Hi @aaronmarkham  Thanks, It works.

**Comment by aaronmarkham** [comment]
Rendering fine:
http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12919/3/install/index.html?platform=MacOS&language=R&processor=CPU

**Comment by anirudhacharya** [comment]
LGTM

**Pull request merged**